assignment6/solve.py

```diff
@@ -0,0 +1,116 @@
+        and reverse_index for items current_array[ind1] and
+        current_array[ind2].
+        """
+        moves.append((ind1, ind2))
+        print 'move from {} to {}'.format(ind1, ind2)
+        current_array[ind1], current_array[ind2] = current_array[ind2], current_array[ind1]
+
+        reverse_index[current_array[ind1]], reverse_index[current_array[ind2]] = \
+            reverse_index[current_array[ind2]], reverse_index[current_array[ind1]]
+
+    current_array = input_array[:]
+    input_size = len(input_array)
+    reverse_index = [0] * input_size
+    for index, element in enumerate(input_array):
+        reverse_index[element] = index
+
+    for index in xrange(input_size):
+        if current_array[index] == result_array[index]:
+            continue
+        if current_array[index] != 0:
+            swap_positions(current_array, reverse_index,
+                           index, reverse_index[0])
+        swap_positions(current_array, reverse_index, reverse_index[result_array[index]], index)
+    return moves
+
+
+def data_from_file(filename):
+    """
+    Get two arrays from .txt file
+    First line contains n space separated integers - first array
+    Second line contains n space separated integers - second array
+    """
+    with open(filename, 'r') as f:
+        def read_array():
+            return [int(item) for item in f.readline().split()]
+        input_array = read_array()
+        result_array = read_array()
+    return input_array, result_array
+
+
+def make_moves(input_array, moves):
+    """
+    Function applies moves to input array if possible
+    For each move (from, to) in moves list functions tries to swap
+    list elements at positions from and to.
+    Value of item at positions with index to must be equal to 0.
+    Function returns array after applying moves from moves array to it.
+    """
+    current_array = input_array[:]
+    for move in moves:
+        assert current_array[move[1]] == 0
+        current_array[move[0]], current_array[move[1]] = \
+            current_array[move[1]], current_array[move[0]]
+    return current_array
+
+
+class TestSolver(unittest.TestCase):
+
+    def test_sample(self):
+        input_array, result_array = data_from_file('inputs/sample_test.txt')
+        moves = solve(input_array, result_array)
+        out_array = make_moves(input_array, moves)
+        self.assertListEqual(result_array, out_array)
+
+    def test_equal(self):
+        input_array, result_array = data_from_file('inputs/equal_test.txt')
+        moves = solve(input_array, result_array)
+        out_array = make_moves(input_array, moves)
+        self.assertListEqual(result_array, out_array)
+
+    def test_random(self):
+        input_array, result_array = data_from_file('inputs/random_test.txt')
+        moves = solve(input_array, result_array)
+        out_array = make_moves(input_array, moves)
+        self.assertListEqual(result_array, out_array)
+

```

Comment:
I am not sure if unit test should have test cases in some external files. Usually unit tests should be very quick so that they can be run before each build so they should be kept as simple as possible and also fast. I think keeping test data in file is making unneccesary overhead on reading file and also it makes test case fail in case of file reading problems. So test case relaying on files is not independent. It may be probably neccesary to keep test data in files in case of very big set of data, but that is not the case here. So what is the advantage of using this files?